src/libpriv/rpmostree-core.c

```diff
@@ -2438,63 +2556,76 @@ rpmostree_context_assemble_tmprootfs (RpmOstreeContext      *self,
+        return FALSE;
+    }
 
-    for (guint i = 0; i < package_list->len; i++)
-      {
-        DnfPackage *pkg = package_list->pdata[i];
-        g_autofree char *cachebranch = rpmostree_get_cache_branch_pkg (pkg);
-        g_autofree char *cached_rev = NULL;
-        gboolean sepolicy_matches;
+  for (guint i = 0; i < overlays->len; i++)
+    {
+      DnfPackage *pkg = overlays->pdata[i];
+      g_autofree char *cachebranch = rpmostree_get_cache_branch_pkg (pkg);
+      g_autofree char *cached_rev = NULL;
+      gboolean sepolicy_matches;
 
-        if (!ostree_repo_resolve_rev (pkgcache_repo, cachebranch, FALSE,
-                                      &cached_rev, error))
-          return FALSE;
+      if (!ostree_repo_resolve_rev (pkgcache_repo, cachebranch, FALSE,
+                                    &cached_rev, error))
+        return FALSE;
 
-        if (self->sepolicy)
-          {
-            if (!commit_has_matching_sepolicy (pkgcache_repo, cached_rev,
-                                               self->sepolicy, &sepolicy_matches,
-                                               error))
-              return FALSE;
+      if (self->sepolicy)
+        {
+          if (!commit_has_matching_sepolicy (pkgcache_repo, cached_rev,
+                                             self->sepolicy, &sepolicy_matches,
+                                             error))
+            return FALSE;
 
-            /* We already did any relabeling/reimporting above */
-            g_assert (sepolicy_matches);
-          }
+          /* We already did any relabeling/reimporting above */
+          g_assert (sepolicy_matches);
+        }
 
-        if (!checkout_pkg_metadata_by_dnfpkg (self, pkg, cancellable, error))
-          return FALSE;
+      if (!checkout_pkg_metadata_by_dnfpkg (self, pkg, cancellable, error))
+        return FALSE;
 
-        if (!add_to_transaction (self, ordering_ts, pkg,
-                                 noscripts, self->ignore_scripts,
-                                 cancellable, error))
-          return FALSE;
+      if (!rpmts_add_install (self, ordering_ts, pkg,
+                              noscripts, self->ignore_scripts,
+                              cancellable, error))
+        return FALSE;
 
-        g_hash_table_insert (pkg_to_ostree_commit, g_object_ref (pkg), g_steal_pointer (&cached_rev));
+      g_hash_table_insert (pkg_to_ostree_commit, g_object_ref (pkg), g_steal_pointer (&cached_rev));
 
-        if (strcmp (dnf_package_get_name (pkg), "filesystem") == 0)
-          filesystem_package = g_object_ref (pkg);
-      }
-  }
+      if (strcmp (dnf_package_get_name (pkg), "filesystem") == 0)
+        filesystem_package = g_object_ref (pkg);
+    }
 
   { DECLARE_RPMSIGHANDLER_RESET;
     rpmtsOrder (ordering_ts);
   }
 
-  rpmostree_output_task_begin ("Overlaying");
+  rpmostree_output_task_begin ("Applying overrides and overlays");
```

Comment:
Can we tweak this to only show `overrides` if there are any?  Maybe something like:

```
if (n_overrides > 0 && n_overlays > 0)
  task_begin ("Applying %u overrides, %u overlays", n_overrides, n_overlays)
else if (n_overlays > 0)
  task_begin ("Applying %u overlays", n_overlays);
else
  task_begin ("Applying %u overrides", n_overrides")
```
?